exporters/prometheus/src/main/java/io/opentelemetry/exporters/prometheus/MetricAdapter.java

```diff
@@ -0,0 +1,191 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.prometheus;
+
+import static io.prometheus.client.Collector.doubleToGoString;
+
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.LongPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import io.prometheus.client.Collector;
+import io.prometheus.client.Collector.MetricFamilySamples;
+import io.prometheus.client.Collector.MetricFamilySamples.Sample;
+import io.prometheus.client.Collector.Type;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Util methods to convert OpenTelemetry Metrics data models to Prometheus data models.
+ *
+ * <p>Each OpenTelemetry {@link MetricData} will be converted to a Prometheus {@link
+ * MetricFamilySamples}, and each {@code Point} of the {@link MetricData} will be converted to
+ * Prometheus {@link Sample}s.
+ *
+ * <p>{@code DoublePoint}, {@code LongPoint} will be converted to a single {@link Sample}. {@code
+ * Summary} will be converted to two (sum and count) plus the number of Percentile values {@code
+ * Sample}s
+ *
+ * <p>Please note that Prometheus Metric and Label name can only have alphanumeric characters and
+ * underscore. All other characters will be sanitized by underscores.
+ */
+final class MetricAdapter {
+
+  static final String SAMPLE_SUFFIX_COUNT = "_count";
+  static final String SAMPLE_SUFFIX_SUM = "_sum";
+  static final String LABEL_NAME_QUANTILE = "quantile";
+
+  // Converts a MetricData to a Prometheus MetricFamilySamples.
+  static MetricFamilySamples toMetricFamilySamples(MetricData metricData) {
+    Descriptor descriptor = metricData.getDescriptor();
+    String fullName =
+        toMetricFullName(
+            descriptor.getName(), metricData.getInstrumentationLibraryInfo().getName());
+    Type type = toMetricFamilyType(descriptor.getType());
+
+    return new MetricFamilySamples(
+        fullName,
+        type,
+        descriptor.getDescription(),
+        toSamples(fullName, descriptor, metricData.getPoints()));
+  }
+
+  private static String toMetricFullName(
+      String descriptorMetricName, String instrumentationLibraryName) {
+    if (instrumentationLibraryName.isEmpty()) {
+      return Collector.sanitizeMetricName(descriptorMetricName);
+    }
+
+    return Collector.sanitizeMetricName(instrumentationLibraryName + "." + descriptorMetricName);
```

Comment:
Given the Javadoc at the beginning of the class, I would expect a join by an underscore.
Furthermore, one less string replace :wink: 
```suggestion
    return Collector.sanitizeMetricName(instrumentationLibraryName + "_" + descriptorMetricName);
```